presto-main/src/main/java/com/facebook/presto/metadata/PolymorphicScalarFunction.java

```diff
@@ -124,10 +132,17 @@ private boolean matchesParameterAndReturnTypes(Method method, List<Type> resolve
                 "method %s has not enough arguments: %s (should have at least %s)", method.getName(), method.getParameterCount(), resolvedTypes.size());
 
         Class<?>[] methodParameterJavaTypes = method.getParameterTypes();
-        for (int i = 0; i < resolvedTypes.size(); ++i) {
-            if (!methodParameterJavaTypes[i].equals(getNullAwareContainerType(resolvedTypes.get(i).getJavaType(), nullableArguments.get(i)))) {
+        for (int i = 0, currMethodArg = 0; i < resolvedTypes.size(); ++i) {
+            Class<?> type = resolvedTypes.get(i).getJavaType();
+            if (nullableArguments.get(i)) {
```

Comment:
This is not true actually. For example, Void without @SqlNullable doesn't work! Because if requires it to be with SqlNullable attribute.

Also, I'm not sure about other possible dependencies.